manifests/params.pp

```diff
@@ -171,6 +171,30 @@
       $psql_path              = pick($psql_path, '/usr/bin/psql')
     }
 
+    'Gentoo': {
+      $user                = pick($user, 'postgres')
+      $group               = pick($group, 'postgres')
+
+      $client_package_name  = pick($client_package_name, 'UNSET')
```

Comment:
I couldn't persuade puppet (4.6) do the right thing with `pick_default(..., undef)`.

```puppet
class client (
  $package_name   = $postgresql::params::client_package_name,
) {
   if $package_name {                                                 
     package { 'postgresql-client':
       name   => $package_name,
     }
   }
```
This always lead the package to be included in the catalog with a null name. I'm guessing this has something to do with the way it's passed through the default argument from `params.pp`?

I've seen the $x='UNSET', if $x != 'UNSET' idiom used in relation to `params.pp` in the past.

While we're on the subject, can anyone tell me how this pattern is supposed to be working?
```puppet
$server_package_name  = pick($server_package_name, 'postgresql')
```
If I understand the variable scoping correctly `$server_package_name` will never have a value before params.pp is evaluated (unless set in top or node scope, neither of which is considered best practice), so the first case of the `pick()` expression should always be null. And if it did have a value, this line of code would be attempting to change the value of a variable which is forbidden because all variables in puppet DSL are really constants? If anyone wants to override the value of these variables, they would do so one the `postgres::server` or `postgres::globals` classes, so is this left over from a refactoring, or have I missed something about how this really works?